system-tests/lib/performance-reporter.js

```diff
@@ -0,0 +1,81 @@
+const path = require('path')
+const chalk = require('chalk')
+const Libhoney = require('libhoney')
+
+const pkg = require('@packages/root')
+const ciProvider = require('@packages/server/lib/util/ci_provider')
+const { commitInfo } = require('@cypress/commit-info')
+
+class StatsdReporter {
+  constructor (runner) {
+    if (!process.env.HONEYCOMB_API_KEY) {
+      return
+    }
+
+    console.log(chalk.green('Reporting to honeycomb'))
+
+    let branch; let commitSha
+
+    this.honey = new Libhoney({
+      dataset: 'systemtest-performance',
+      writeKey: process.env.HONEYCOMB_API_KEY,
+    })
+
+    commitInfo().then((commitInformation) => {
+      const ciInformation = ciProvider.commitParams() || {}
+
+      this.branch = commitInformation.branch || ciInformation.branch
+      this.commitSha = commitInformation.sha || ciInformation.sha
+    })
+
+    runner.on('test', (test) => {
+      test.wallclockStart = Date.now()
+    })
+
+    runner.on('test end', (test, done) => {
+      // Skipped tests never get a 'start' event, but they still get 'test end' somehow.
+      if (!test.state || test.state === 'skipped') {
+        return
+      }
+
+      const title = test.titlePath().join(' / ')
+      // This regex pulls apart a string like `e2e async timeouts / failing1 [electron]`
+      // into `e2e async timeouts / failing1` and `electron`, letting us use the same
+      // test name for all browsers, with the browser as a separate field.
+      // The browser capture group is optional because some tests aren't browser specific,
+      // in which case it will be undefined and not passed as a field to honeycomb.
+      const [, testTitle, browser] = title.match(/(.+?)(?: \[([a-z]+)\])?$/)
```

Comment:
FYI: If this doesn't match for whatever reason, if we change the structure of the suffix we're matching on or something, it will throw b/c `.match` will return null.

Reply:
If the second part fails to match, because it's optional the /.+/ will consume the entire string. For example,

`"foobar (firefox)".match(/(.+?)(?: \[([a-z]+)\])?$/)`

returns `["foobar (firefox)", "foobar (firefox)"]` rather than throwing.

Reply:
Ah right 🤦 - all good then